Implement Mayhem

nottirb opened the pull request:
- Added a `Dockerfile` for fuzz testing
- Added `Mayhemfile(s)` for fuzz testing
- Added a `mayhem.yml` file for automatically running fuzz tests with Mayhem using the mcode action
- Wrote 3 fuzz harnesses

xansec commented on fuzz/fuzz_targets/rabbit.rs:
```diff
@@ -0,0 +1,60 @@
+use honggfuzz::fuzz;
```
Asking again for confirmation of fixed random numbers dependent on test case/seed.

xansec commented on mayhem/hc-256.Mayhemfile:
```diff
@@ -0,0 +1,10 @@
+project: stream-ciphers
```
Please omit image name.

xansec commented on mayhem/salsa20.Mayhemfile:
```diff
@@ -0,0 +1,10 @@
+project: stream-ciphers
```
Please omit image name.

xansec reviewed (changes requested):
Please make the requested changes.

nottirb commented:
I removed the image name from the Mayhemfiles, and replaced StdRng with Pcg64 to ensure that it generates the same random number on different builds/platforms.

xansec reviewed (approved):
LGTM!
